Lib/email/feedparser.py

```diff
@@ -534,3 +559,85 @@ class BytesFeedParser(FeedParser):
 
     def feed(self, data):
         super().feed(data.decode('ascii', 'surrogateescape'))
+
+
+class NonIdentityTransferDecodingFeedParser(abc.ABC):
+    """
+    This is an abstract base class; only its subclasses should be instantiated
+     directly.
+
+    Instances of this class work like FeedParser except that the feed() method,
+     prior to parsing the input, transparently decodes the input consistent with
+     RFC 2045, s. 6.2.  Each concrete subclass reverses one of the non-identity
+     Content-Transfer-Encoding transformations described there.
+    """
+
+    def __init__(self, *args, **kwargs):
+        self._bytes_feed_parser = BytesFeedParser(*args, **kwargs)
+
+    @abc.abstractmethod
+    def feed(self, text):
+        pass
+
+    def close(self):
+        return self._bytes_feed_parser.close()
+
+
+class Base64TransferDecodingFeedParser(NonIdentityTransferDecodingFeedParser):
+    """
+    Concrete, "base64" implementation of TransferDecodingFeedParser.
```

Comment:
This docstring, and the docstring for `QuotedPrintableTransferDecodingFeedParser` seem redundant. Seeing as the class and base class names are quite descriptive, maybe just provide an RFC (or other) citation in the docstrings?

Reply:
Thanks, I incorporated the recommendation in a massive update I'm about to apply.